transformer_engine/common/fused_attn/fused_attn_f16_arbitrary_seqlen.h

```diff
@@ -14,53 +14,111 @@
-                                      float p_dropout, NVTE_QKV_Layout qkv_layout,
-                                      NVTE_Bias_Type bias_type, NVTE_Mask_Type mask_type,
-                                      const Tensor *input_QKV, const Tensor *input_Bias,
-                                      Tensor *output_O, NVTETensorPack *Aux_CTX_Tensors,
-                                      const Tensor *cu_seqlens, const Tensor *rng_state,
-                                      Tensor *workspace, cudaStream_t stream, cudnnHandle_t handle);
+namespace fused_attn {
+void fused_attn_arbitrary_seqlen_fwd_impl(
+                int64_t b, int64_t h, int64_t hg, int64_t s_q, int64_t s_kv, int64_t d,
+                bool is_training, float scaling_factor, float dropout_probability,
+                NVTE_QKV_Layout layout,
+                NVTE_Bias_Type bias_type, NVTE_Mask_Type mask_type,
+                void *devPtrQ, void *devPtrK, void *devPtrV,
+                void *devPtrSoftmaxStats, void *devPtrO,
+                void* devPtrDropoutSeed, void* devPtrDropoutOffset,
+                void* devPtrCuSeqlensQ, void* devPtrCuSeqlensKV,
+                cudnn_frontend::DataType_t tensorType,
+                void *workspace, size_t *workspace_size,
+                cudaStream_t stream, cudnnHandle_t handle, bool* check_support);
 
-void fused_attn_arbitrary_seqlen_bwd_qkvpacked(size_t batch, size_t max_seqlen, size_t num_head,
-                                      size_t head_dim, float attn_scale, float p_dropout,
-                                      NVTE_QKV_Layout qkv_layout, NVTE_Bias_Type bias_type,
-                                      NVTE_Mask_Type mask_type, const Tensor *input_QKV,
-                                      const Tensor *input_O,
-                                      const Tensor *input_dO, Tensor *output_S,
-                                      Tensor *output_dQKV, Tensor *output_dBias,
-                                      const Tensor *cu_seqlens, const Tensor *rng_state,
-                                      Tensor *workspace, cudaStream_t stream, cudnnHandle_t handle);
+void fused_attn_arbitrary_seqlen_bwd_impl(
+                int64_t b, int64_t h, int64_t hg, int64_t s_q, int64_t s_kv, int64_t d,
+                float scaling_factor, float dropout_probability, NVTE_QKV_Layout layout,
+                NVTE_Bias_Type bias_type, NVTE_Mask_Type mask_type,
+                void* devPtrQ, void* devPtrKTranspose, void* devPtrVTranspose,
+                void* devPtrO, void* devPtrSoftmaxStats,
+                void* devPtrdQ, void* devPtrdK, void* devPtrdV, void* devPtrdO,
+                void* devPtrDropoutSeed, void* devPtrDropoutOffset,
+                void* devPtrCuSeqlensQ, void* devPtrCuSeqlensKV,
+                cudnn_frontend::DataType_t tensorType, void *workspace, size_t *workspace_size,
+                cudaStream_t stream, cudnnHandle_t handle, bool* check_support);
+}  // namespace fused_attn
```

Comment:
Why do we need to expose implementation to the header? I supposed the header should only contain the public API, doesn't the implementation only be used in `fused_attn_f16_arbitrary_seqlen.cu`?

Reply:
This is for the `check_support` feature mentioned below.